apis/nucleus/src/components/listbox/QlikSwitchButton/QlikSwitch.jsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { makeStyles } from '@material-ui/core';
+import Switch from '@material-ui/core/Switch';
+
+export default function QlikSwitch({ checked, onChange, iconOn, iconOff }) {
+  const styles = makeStyles((theme) => ({
```

Comment:
I suggest to move `makeStyles()` outside the React component and pass the props to `useStyles(props)`. The [docs](https://mui.com/styles/basics/#adapting-based-on-props) explains in more detail how to do it.

Moving to MUI5 would require this to be written in either `sx` or `styled()`. It would be interesting to try to write this with styled.

Reply:
I wrote this with `styled` to start with but then we got the problem with the prefixes not being added, which is why I created https://github.com/qlik-oss/nebula.js/pull/768 . Then, we did not want that workaround, and I was advised to use `makeStyles`. So yeah the loop has been completed 😄  
I could potentially break it out but I spent significant time on adapting the styles to makeStyles so feel a bit hesitant to change this again.